### PR TITLE
[Swift] fix URLSession file upload

### PR DESCRIPTION
**Pull request by 4brunu**
The URLSession implementation was failing to upload files to the server.
The Alamofire implementation it's working.

In this PR the URLSession implementation is fixed.

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

### CODE REVIEW

**Comment by wing328** [comment]
@4brunu thanks for the fix. What about adding a test with the `uploadFile` in the Petstore API to ensure the file upload works moving forward?

**Comment by 4brunu** [comment]
Hey, I added some unit tests as you suggested 🙂

File: samples/client/petstore/swift5/alamofireLibrary/SwaggerClientTests/SwaggerClientTests/FileUtils.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//
+//  FileUtils.swift
+//  SwaggerClientTests
+//
+//  Created by Bruno Coelho on 11/03/2020.
+//  Copyright © 2020 Swagger. All rights reserved.
```
Comment (wing328) [inline samples/client/petstore/swift5/alamofireLibrary/SwaggerClientTests/SwaggerClientTests/FileUtils.swift]: Please use `Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)` instead.

Comment (4brunu) [reply samples/client/petstore/swift5/alamofireLibrary/SwaggerClientTests/SwaggerClientTests/FileUtils.swift]: Like this?

Comment (wing328) [reply samples/client/petstore/swift5/alamofireLibrary/SwaggerClientTests/SwaggerClientTests/FileUtils.swift]: Yup

**Comment by jerbob92** [comment]
@4brunu I'm not sure why this works for you, but the changes you did are not according to the multipart spec: https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html

```
Note that the encapsulation boundary must occur at the beginning of a line, i.e., following a CRLF, and that that initial CRLF is considered to be part of the encapsulation boundary rather than part of the preceding part. The boundary must be followed immediately either by another CRLF and the header fields for the next part, or by two CRLFs, in which case there are no header fields for the next part (and it is therefore assumed to be of Content-Type text/plain).
NOTE: The CRLF preceding the encapsulation line is considered part of the boundary so that it is possible to have a part that does not end with a CRLF (line break). Body parts that must be considered to end with line breaks, therefore, should have two CRLFs preceding the encapsulation line, the first of which is part of the preceding body part, and the second of which is part of the encapsulation boundary.

The requirement that the encapsulation boundary begins with a CRLF implies that the body of a multipart entity must itself begin with a CRLF before the first encapsulation line -- that is, if the "preamble" area is not used, the entity headers must be followed by TWO CRLFs. This is indeed how such entities should be composed. A tolerant mail reading program, however, may interpret a body of type multipart that begins with an encapsulation line NOT initiated by a CRLF as also being an encapsulation boundary, but a compliant mail sending program must not generate such entities.

Encapsulation boundaries must not appear within the encapsulations, and must be no longer than 70 characters, not counting the two leading hyphens.
```
I read it like the format should be:

{--boundary} {newline} {header1} {newline} {header2} {newline} {newline} {body} {newline} {--boundary} {newline} {header1} {newline} {header2} {newline} {newline} {body} {newline} {--boundary--} {newline}

This is also what other http clients do that I reviewed. 
However, your change does this:

{--boundary} {newline} {header1} {newline} {newline} {header2} {newline} {newline} {body} {newline} {newline}  {--boundary} {newline} {header1} {newline} {newline} {header2} {newline} {newline} {body} {newline} {newline} {--boundary--}

The differences are:
 - One newline too much between headers
 - One newline too much after the body

The multipart form data parser in Go (our backend) does not parse your generated body. When I change it to mine it does work properly. 

The best way to compare is to use two generated clients, one with URLSession and one with Alamofire, and then post the form data to a RequestBin to review the differences.

If you want I can create a pull request.

**Comment by 4brunu** [comment]
@jerbob92 can you please create a PR so I can test them on my side?
From what I read in your comment, I think you are right.

**Comment by jerbob92** [comment]
@4brunu since I would have to do some stuff before I can submit the PR, I didn't yet.
You can review the changes in my branch: https://github.com/jerbob92/openapi-generator/tree/feature/swift-urlsession-file-upload

If you think that's a good fix I will create a PR.

**Comment by 4brunu** [comment]
@jerbob92 I just tested your branch and it works.
Could you please submit a PR with that change please?
Thanks

**Pull request merged**